### PR TITLE
Add initial automation framework skeleton

### LINKED ISSUE #1: Create initial automation framework skeleton
As a Quality Engineer, I want to be able to run integrade tests using py.test so that cloudigrade and other related projects can be tested.

## Acceptance Criteria 

- [x] Verify that integrade is a Python package and it and its required packages can be installed using pip
- [x] Verify that integrade have a Makefile with make targets for testing the automation framework and linting the code.
- [x] Verify that integrade have a `integrade.tests` package where all the integration tests will be placed 
- [x] Verify that integrade have a tests directory for its own unit tests
- [x] Verify that integrade is being tested (its own unit tests) and linting on Travis
- [x] Verify that integrade code coverage is being collected on codecov
- [x] Verify that integrade have documentation about how to install it and run the integration tests.

### PR DESCRIPTION
**Pull request by elyezer**
Close #1

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+No matter if you are interested on contributing to new functional tests for
+cloudigrade projects or improving integrade automation framework, make sure
+that the tests pass and also the code follows the code style properly. The
+``make lint`` can help you linting your code and ensuring the code style.
+
+Configuring and Running
+=======================
+
+To be done... integrade is justing starting and in the future you can see how
```
Comment (ilyavw) [inline README.rst]: `integrade is justing starting` -> `integrade is just starting`

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+lint:
+	flake8 .
+
+test:
+	py.test tests
+
+test-coverage:
+	py.test --verbose --cov-report term --cov=integrade tests
+
+.PHONY: all install install-dev lint test test-coverage
```
Comment (infinitewarp) [inline Makefile]: Oooh, this is neat! I didn't know about `.PHONY` and had to look it up. If I understand correctly [what I read](https://stackoverflow.com/a/2145605), this allows these commands to work even if there's a file with the same name. I may have to steal this for one of our other make files.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+to configure and run the functional tests on this section.
```
Comment (ilyavw) [inline README.rst]: **in** this section

File: README.rst (hunk 1 above)
Comment (elyezer) [reply README.rst]: Doh

File: Makefile (hunk 2 above)
Comment (elyezer) [reply Makefile]: Yeah, that is correct. I am glad you found it useful.

File: README.rst (hunk 3 above)
Comment (elyezer) [reply README.rst]: the old in/on/at problem from non-native English speakers... thank you!

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	pip install -e .[dev]
+
+lint:
+	flake8 .
+
+test:
+	py.test tests
+
+test-coverage:
+	py.test --verbose --cov-report term --cov=integrade tests
```
Comment (ilyavw) [inline Makefile]: Seems like there is an issue parsing the coverage report, are you sure its configured correctly?

https://codecov.io/github/cloudigrade/integrade/commit/ff0920a01e68e2db2cfb4358fa9681b0356661b7

Comment (elyezer) [reply Makefile]: Maybe because there is no line of code yet? I will double check

Comment (ilyavw) [reply Makefile]: Perhaps, but I'd expect your tests to be included in coverage too.

Comment (elyezer) [reply Makefile]: It seems that the report being empty is the issue. The page you've linked says:

```
There was an error processing coverage reports.

Most common issues

File paths are not resolving.  Reports are empty.  Report has expired.
```

Comment (ilyavw) [reply Makefile]: Running `make test-coverage` I get a non empty file, but it seems like it only looks at the `__init__` files. Your test file isn't being picked up, maybe you need a `coveragerc` file to define your sources?

Comment (elyezer) [reply Makefile]: That test file is for the framework and we don't want to measure coverage for it. We will be looking for code coverage only for automation modules, ones inside integrade.* but excluding everything integrade.tests.*

Comment (ilyavw) [reply Makefile]: Generally those files are included too (see https://codecov.io/gh/cloudigrade/cloudigrade/tree/master/cloudigrade/account/tests) not as code that is tested, but code that is run. Goal is to make sure you don't have any code in your tests that doesn't actually get run.

Comment (elyezer) [reply Makefile]: Makes sense, lemme try adding the /tests package then

**Pull request merged**